src/accelerate/utils/operations.py

```diff
@@ -468,13 +468,11 @@ def _is_fp16_bf16_tensor(tensor):
     return recursively_apply(_convert_to_fp32, tensor, test_type=_is_fp16_bf16_tensor)
 
 
-class ConvertOutputsToFp32:
+def convert_outputs_to_fp32(model_forward):
     """
     Decorator to apply to a function outputing tensors (like a model forward pass) that ensures the outputs in FP16
     precision will be convert back to FP32.
 
-    Use a class instead of a decorator because otherwise, the prepared model can no longer be pickled (issue #273).
-
```

Comment:
Are we sure this does not make a regression with this issue?

Reply:
The issue will always exist because it can never be pickled if we've wrapped it in a mixed precision decorator. So it will always have to go through `unwrap_model`. And this only ever comes up if we do mixed precision. See the nb shown in the bug report here: https://colab.research.google.com/drive/11fvrk1Jslw2VIRTF6h0pdGgJx5AkAIMv?usp=sharing

Where specifically our wrapping around `torch.cuda.amp` causes issues.

Reply:
I can try to see if it's possible to inject a warning before trying to pickle/save it and tell the user they need to unwrap the model first.